### ISSUE TITLE
CODEOWNERS malformed lines are silently ignored

### ISSUE THREAD

**Issue by chrismlittle123** [chrismlittle123/check-my-toolkit#114]
## Description

When parsing CODEOWNERS files, lines that contain a pattern but no owners (malformed entries) are silently ignored. No violation or warning is reported.

## Reproduction

CODEOWNERS file:
```
*
/docs @owner
```

The first line `*` is a pattern without any owners - this is malformed.

check.toml:
```toml
[process.codeowners]
enabled = true

[[process.codeowners.rules]]
pattern = "/docs"
owners = ["@owner"]
```

Running `cm process check` passes without reporting the malformed line.

## Location

`src/process/tools/codeowners.ts` (lines 97-126)

```typescript
private parseCodeownersLine(line: string, lineNumber: number): ParsedRule | null {
  const tokens = line.split(/\s+/).filter(Boolean);

  if (tokens.length < 2) {
    // Invalid line - pattern without owners
    return null;  // Silently returns null
  }
  // ...
}
```

## Expected Behavior

Malformed lines should be reported as violations or warnings, not silently skipped. This helps users catch configuration errors in their CODEOWNERS files.

## Suggested Fix

Instead of returning `null`, add a violation:

```typescript
if (tokens.length < 2) {
  this.addViolation({
    rule: `${this.rule}.malformed`,
    tool: this.toolId,
    message: `Line ${lineNumber}: Pattern "${tokens[0]}" has no owners`,
    severity: "warning",
  });
  return null;
}
```

## Environment

- check-my-toolkit version: 1.3.0
